sklearn/utils/_param_validation.py

```diff
@@ -173,12 +173,12 @@ class _InstancesOf(_Constraint):
 
     Parameters
     ----------
-    type : type
-        The valid type.
+    param_type : param_type
+        The valid paramater type.
     """
 
-    def __init__(self, type):
-        self.type = type
+    def __init__(self, param_type):
+        self.param_type = param_type
```

Comment:
As commented in https://github.com/scikit-learn/scikit-learn/issues/23599#issuecomment-1153187862:

The "type" here is not the builtin `type` anymore but the locally scoped "type". A simple Python example:

```python
class A:
    def __init__(self, type):
        self.type = type
        
a = A(100)
print(a.type)
# 100
```

Reply:
Thank you for clarifiying